src/App.tsx

```diff
@@ -1,24 +1,155 @@
+
+  useEffect(() => {
+    client.get<Todo[]>(`/todos?userId=${USER_ID}`)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessege(ErrorMessage.CANNOT_LOAD_TODOS);
+      });
+  }, []);
+
+  const addTodo = (title: string) => {
+    setIsLoading(true);
+    setErrorMessege(ErrorMessage.NONE);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    client.post<Todo>('/todos', {
+      userId: USER_ID,
+      title,
+      completed: false,
+    })
+      .then(newTodo => {
+        setTodos((currentTodos) => [...currentTodos, newTodo]);
+      })
+      .catch((error) => {
+        setErrorMessege(ErrorMessage.UNABLE_TO_ADD_A_TODO);
+        throw error;
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setErrorMessege(ErrorMessage.NONE);
+    setSelectedId(todoId);
+
+    client.delete(`/todos/${todoId}`)
+      .then(() => setTodos(todos.filter((todo) => todo.id !== todoId)))
+      .catch(() => {
+        setErrorMessege(ErrorMessage.UNABLE_TO_DELETE_A_TODO);
+      })
+      .finally(() => {
+        setSelectedId(-1);
+      });
+  };
+
+  const deleteAllCompleted = () => {
+    setIsLoadingCompleted(true);
+
+    const completedTodos = todos.filter((todo) => todo.completed);
+
+    const deletePromises = completedTodos.map((todo) => client.delete(`/todos/${todo.id}`));
+
+    Promise.all(deletePromises)
+      .then(() => {
+        setTodos(currentTodo => currentTodo.filter((todo) => !todo.completed));
+      })
+      .catch((error) => {
+        setErrorMessege(ErrorMessage.UNABLE_TO_DELETE_A_TODO);
+        throw error;
+      })
+      .finally(() => {
+        setIsLoadingCompleted(false);
+      });
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    switch (filter) {
+      case FilterType.ACTIVE:
+        return !todo.completed;
+      case FilterType.COMPLETED:
+        return todo.completed;
+      case FilterType.ALL:
+      default:
+        return true;
+    }
+  });
+
+  const itemsLeft = todos.filter((todo) => !todo.completed).length;
+
+  const closeErrorMsg = () => {
+    setErrorMessege(ErrorMessage.NONE);
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          addTodo={addTodo}
+          setErrorMessege={setErrorMessege}
+          isLoading={isLoading}
```

Comment:
```suggestion
          isLoading={!!tempTodo}
```
you can pass here tempTodo